packages/daemon/src/app/controller/base-controller.ts

```diff
@@ -0,0 +1,16 @@
+import { Context, inject } from 'midway';
+
+export class BaseController {
+  @inject()
+  ctx: Context;
+
+  successRes(data: any, code = 200): void {
+    this.ctx.body = data;
+    this.ctx.status = code;
+  }
+
+  failRes(message: string, code = 400): void {
```

Comment:
```suggestion
  fail(message: string, status = 400): void {
```

Reply:
I prefer the type of `message` to be string, in many cases, there is no error throw out.
```js
if (!this.ctx.query.name) {
  fail('no field "name" found', 400);
}
```

Reply:
('no field "name" found') is an error IMO.